### PR TITLE
Laura Vega's DMV

### PR DESCRIPTION
**Pull request by laurarvegav**
Iterations 1 to 3 complete, iteration 4 last item (_Add in a second data source for vehicles and create some Vehicle objects from NY State Registration Data: https://data.ny.gov/resource/w4pv-hbkt.json_) not passing in full the coded testing.

### CODE REVIEW

File: spec/facility_spec.rb [hunk 1]
```diff
@@ -1,26 +1,142 @@
...
+  end
+
+  describe "#register a vehicle" do
+    it "can register a vehicle" do
+      @facility_1.add_service('Vehicle Registration')
+      expect(@cruz.registration_date).to eq (nil)
+      expect(@facility_1.registered_vehicles).to eq []
+      expect(@facility_1.collected_fees).to eq(0)
+
+      @facility_1.register_vehicle(@cruz)
```
Comment (KatBrandt) [inline spec/facility_spec.rb:65]: Yes for the white spaces. Feel free to use more white spaces in your code to organize chunks of a test, helps the eyes.

File: spec/facility_spec.rb [hunk 2]
```diff
@@ -1,26 +1,142 @@
...
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+    end
+
+    it 'can collect corresponding fees' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.collected_fees).to eq(100)
+      @facility_1.register_vehicle(@camaro)
+      @facility_1.register_vehicle(@bolt)
+      expect(@facility_1.collected_fees).to eq(325)
```
Comment (KatBrandt) [inline spec/facility_spec.rb:77]: Great job testing the accumulation of these fees.

File: spec/facility_spec.rb [hunk 3]
```diff
@@ -1,26 +1,142 @@
...
+    end
+  end
+  describe "#issue drivers license and guarantee requirements are met" do
+    it "guarantees requirements to administer written test are met" do
+      expect(@registrant_1.license_data).to eq ({written: false, license: false, renewed: false})
+      expect(@registrant_1.permit?).to eq (true)
```
Comment (KatBrandt) [inline spec/facility_spec.rb:83]: typically we do not leave a space between the eq or the method and the opening parenthesis. I've seen this a lot in your code, so general rule, no space between the opening parenthsis and whatever comes before it (if, elsif, method name, eq, etc)

File: spec/vehicle_spec.rb [hunk 4]
```diff
@@ -33,4 +34,21 @@
       expect(@camaro.electric_vehicle?).to eq(false)
     end
   end
+
+  describe'#assign_plate_type' do
```
Comment (KatBrandt) [inline spec/vehicle_spec.rb:38]: Nice! You do have this test in here! Great.

**Review by KatBrandt — COMMENTED** [review]
Overall, great job! It's clear you understand the concepts that we've covered so far. A general comment, in your tests, use more white spaces, between the it blocks and within a test to show different sections of the test, it will make it easier to read.

In regards to your question about over-committing, I do think 14 branches and 119 commits is a bit too many for a project of this size. Try making branches that encompass a bit more breadth and just commit after a test and after a method is written. Of course I'd rather have you over committing than under committing so this is not a problem! You'll continue to get a feel for it. 

Great work Laura!

File: lib/vehicle_factory.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            model: registration[:model],
+            engine: :ev,
+            county: registration[:county]
+        })
+    end
+
+    def norm_ny_data(registrations)
+       registrations.find_all do |registration|
+            (registration[:"record_type"] == "VEH") 
+        end
```
[the remark this answers is not in the document]
Comment (laurarvegav) [reply lib/vehicle_factory.rb:29]: I wanted to use the data related to vehicles only, not boats/trucks/...

[the remark this answers is not in the document]
Comment (KatBrandt) [reply lib/vehicle_factory.rb:29]: Gotcha! Great job with that!

**Pull request closed without merging**